website/static/js/pages/project-settings-page.js

```diff
@@ -227,7 +227,13 @@ $(document).ready(function() {
       var unchecked = checkedOnLoad.filter('#selectAddonsForm input:not(:checked)');
 
       if(unchecked.length > 0 || checked.length > 0) {
-        return 'The changes on addon setting are not submitted!';
+          return 'The changes on addon setting are not submitted!';
+      }
+    /* Before closing the page, Check whether changes made to category, title or description are updated or not */
+      if (projectSettingsVM.title() !== projectSettingsVM.titlePlaceholder ||
```

Comment:
Hi, Extern here. I'm not sure I'm understanding what you mean here. I tired a couple things though. 
Right now I made a function in the project class as shown below:

`        navigateAway: function() {
            if (projectSettingsVM.title() !== projectSettingsVM.titlePlaceholder ||
                projectSettingsVM.description() !== projectSettingsVM.descriptionPlaceholder ||
                projectSettingsVM.selectedCategory() !== projectSettingsVM.categoryPlaceholder) {
                return 'There are unsaved changes in your project settings.';
            }
        },`

I then call this function with `ProjectSettings.navigateAway();` in the place this check was previously, within the 'beforeunload' function.
As the functionality is no longer what it should be, I don't think I quite understand what you mean. Can you elaborate some? (I'm also not opposed to talking directly!)